climt/_core/constants.py

```diff
@@ -0,0 +1,303 @@
+        print()
+        print('Stellar Constants')
+        print('===========================')
+        self._list_constants(self.constant_categories["stellar_constants"])
+        print()
+        print('Atmospheric Constants')
+        print('===========================')
+        self._list_constants(self.constant_categories["atmospheric_constants"])
+        print()
+        print('Condensible Constants')
+        print('===========================')
+        self._list_constants(self.constant_categories["condensible_constants"])
+        print()
+        print('Oceanographic Constants')
+        print('===========================')
+        self._list_constants(self.constant_categories["oceanographic_constants"])
+        print()
+        print('Miscellaneous Constants')
+        print('===========================')
+        self._list_constants(self.constant_categories["miscellaneous_constants"])
+        print()
+
+    def add_constants_to_library(self, constant_descriptions):
+        """
+        Add new constants to the library.
+
+        Args:
+
+           constant_descriptions (dict):
+               Dictionary containing the description of the constants.
+               The key should be the name of the constant, and the value
+               should be a dictionary containing the following keys:
+
+               * type (string):
+                   Type of constant. Is one of:
+
+                       * :code:`'physical_constants'`,
+                       * :code:`'chemical_constants'`,
+                       * :code:`'planetary_constants'`,
+                       * :code:`'atmospheric_constants'`,
+                       * :code:`'oceanographic_constants'`,
+                       * :code:`'stellar_constants'`,
+                       * :code:`'condensible_constants'`,
+                       * :code:`'miscellaneous_constants'`,
+                       * value (float):
+                           The value assigned.
+               * units (string):
+                   The units of the value, e.g, m/s, J/kg.
+
+        Raises:
+
+           IndexError:
+               If the constant exists in the library or if the type
+               of constant is not valid.
+        """
+
+        for name in constant_descriptions.keys():
+
+            if name in self.current_constants.keys():
+                raise IndexError('{} already present in library.'.format(name))
+            value = constant_descriptions[name]['value']
+            units = constant_descriptions[name]['units']
+            type = constant_descriptions[name]['type']
+
+            if type not in self.constant_categories.keys():
+                raise IndexError('{} is not a valid category.'.format(type))
+
+            self.constant_categories[type] = self.constant_categories[type].union((name,))
+
+            self.current_constants[name] = DataArray(value, attrs={'units': units})
+
+    def modify_constants_in_library(self, constant_descriptions):
```

Comment:
It might be clearer to use a method like modify_constant(name, value, units), and call it sequentially. You may want to have both.

It's probably too verbose to include "in_library" in the name, unless you want to explicitly state you're not altering the Sympl constants. But I'd encourage you to alter them. Actually for that matter, this whole object should just be a module with functions instead of an object, since you only want one constants library to ever exist.